src/ansys/grantami/jobqueue/_models.py

```diff
@@ -138,61 +138,101 @@ def to_dict(self) -> Dict[str, Union[int, None]]:
+        Path to the local file to use in a JobRequest. Can be absolute or relative.
+    virtual_path : str or pathlib.Path
+        Virtual path to use to refer to the file in the JobRequest. Must be relative.
     """
 
-    def __init__(self, file_type: _FileType):
-        self.file_type = file_type
-        self._path: Union[pathlib.Path, None] = None
-        self._id: Optional[str] = None
+    def __init__(self, path: Union[str, pathlib.Path], virtual_path: Union[str, pathlib.Path]):
+        self._path: pathlib.Path = pathlib.Path(path)
+        virtual_path = pathlib.Path(virtual_path)
+        self._validate_virtual_path(virtual_path)
+        self._virtual_path: pathlib.Path = virtual_path
 
-    @classmethod
-    def from_pathlib_path(cls, path: pathlib.Path, file_type: _FileType) -> "_JobFile":
+    @staticmethod
+    def _validate_virtual_path(path: pathlib.Path) -> None:
         """
-        Create a ``JobFile`` object from a ``pathlib.Path`` object.
+        Validate the provided virtual path. It must be relative and within the working directory.
 
         Parameters
         ----------
         path : pathlib.Path
-            Path to the file.
-        file_type : _FileType
-            Type of file to create.
+            Path to validate.
+        """
+        if path.is_absolute():
+            raise ValueError("Virtual path must be relative.")
+        if not path.resolve().is_relative_to(os.getcwd()):
+            raise ValueError("Virtual path must be safe.")
+
+    @property
+    def path(self) -> pathlib.Path:
+        """
+        Path to the local file.
 
         Returns
         -------
-        _JobFile
-            Created ``JobFile`` object.
+        pathlib.Path
+            Path to the local file.
         """
-        new_obj = cls(file_type)
-        new_obj.path = path
-        return new_obj
+        return self._path
+
+    @property
+    def virtual_path(self) -> pathlib.Path:
+        """
+        Virtual path in the JobRequest context.
+
+        Returns
+        -------
+        pathlib.Path
+            Path to use in the context of the request.
+        """
+        return self._virtual_path
+
+
+class _JobFile:
+    """
+    Represents a file associated with a job request.
+
+    Parameters
+    ----------
+    path : pathlib.Path
+        Path to the local file.
+    file_type : _FileType
+        Type of file being represented.
+    """
+
+    def __init__(self, path: pathlib.Path, file_type: _FileType):
+        self.file_type = file_type
+        self._path: pathlib.Path = path
+        self._id: Optional[str] = None
+        self._virtual_path: Optional[pathlib.Path] = None
 
     @classmethod
-    def from_string(cls, file_path: str, file_type: _FileType) -> "_JobFile":
+    def from_job_file(cls, file: JobFile, file_type: _FileType) -> "_JobFile":
         """
-        Create a ``JobFile`` object from a string path.
+        Create a ``JobFile`` object from a ``JobFile`` object.
```

Comment:
We might want to rename this private class for clarity. Not sure what to, though.

```suggestion
        Create a ``_JobFile`` object from a ``JobFile`` object.
```